core/node/api_server/src/tx_sender/whitelist.rs

```diff
@@ -38,6 +38,16 @@ impl TxSink for WhitelistedDeployPoolSink {
         execution_output: &SandboxExecutionOutput,
         validation_traces: ValidationTraces,
     ) -> Result<L2TxSubmissionResult, SubmitTxError> {
+        // Check if the transaction initiator is allowlisted
+        let initiator = tx.initiator_account();
+        if self.shared_allow_list.is_address_allowed(&initiator).await {
```

Comment:
Bikeshedding: IMO, it could make sense to extract a method returning a boolean flag whether to proceed with delegation to `master_pool_sink` to a separate private method, so that the delegation doesn't need to be duplicated.

Reply:
makes sense! i've just added that, let me know your thoughts :)